kolibri/core/tasks/decorators.py

```diff
@@ -0,0 +1,108 @@
+import logging
+from functools import partial
+from importlib import import_module
+
+from django.apps import apps as django_apps
+
+from kolibri.core.tasks.exceptions import FunctionNotRegisteredAsJob
+from kolibri.core.tasks.job import JobRegistry
+from kolibri.core.tasks.job import Priority
+from kolibri.core.tasks.job import RegisteredJob
+from kolibri.core.tasks.utils import stringify_func
+
+
+logger = logging.getLogger(__name__)
+
+
+class _TaskDecorators(object):
+    """
+    Task related decorators.
+    """
+
+    def __init__(self):
+        self.priority = Priority
+
+    def register(
+        self,
+        func=None,
+        job_id=None,
+        validator=None,
+        permission=None,
+        priority=Priority.REGULAR,
+    ):
+        if func is None:
+            return partial(
+                self.register,
+                job_id=job_id,
+                validator=validator,
+                permission=permission,
+                priority=priority,
+            )
+
+        registered_job = RegisteredJob(
+            func,
+            job_id=job_id,
+            validator=validator,
+            permission=permission,
+            priority=priority,
+        )
+
+        # Expose registered_job's api to func.
+        setattr(func, "task", registered_job)
+
+        funcstring = stringify_func(func)
+        JobRegistry.REGISTERED_JOBS[funcstring] = registered_job
+
+        logger.debug("Successfully registered %s as job", funcstring)
+
+        return func
+
+    def config(
+        self,
+        func=None,
+        group=None,
+        cancellable=False,
+        track_progress=False,
+    ):
+        if func is None:
+            return partial(
+                self.config,
+                group=group,
+                cancellable=cancellable,
+                track_progress=track_progress,
+            )
+
+        funcstring = stringify_func(func)
+
+        try:
+            registered_job = JobRegistry.REGISTERED_JOBS[funcstring]
```

Comment:
The main motivation behind separation of `register` and `config` decorator was to separate "configurable" attributes from other attributes. Every kwarg in one decorator will make that decorator too longish and kind of hard to read, don't you think?

Reply:
I think there are two alternatives:

1. Just be ok with lots of kwargs - this feels OK, especially as it has the advantage of the function parameters being explicit.
2. Add a `config` kwarg that accepts a dict of the configurable attributes - this clearly delineates these attributes as 'configurable' but you lose the function parameter introspectability to know what arguments can be passed.

I think because I don't really see the kwargs being passed to `register` and `config` right now as being that different to each other (for example, if a task is cancellable is not really configurable, and more a matter of whether the internal implementation of the task properly handles when a cancel event is raised) my preference is for what I suggested above, which is 1.

It's possible we may have a clearer need for separating configuration from registration in the future, but for now, it seems unnecessary.

Reply:
Okay, so let us move forward with 1 as said above. 

For `POST /api/tasks/`, I added `@task.permission_classes([])` as a separate decorator, I think we will be fine with that or do we want to keep `permission_classes` as a kwarg in register? I will say separation will be much better in this case to allow for easy readability of permissions especially when bitwise operators are used.

Reply:
I think to maintain simplicity, just adding it as another kwarg to register for now would be fine - we can then see if it produces monstrous code and we need to rethink, but I think it is mostly just a matter of code formatting, so with `black` running on the code, I think it should be readable.

Reply:
Okay, let's go with only register. One to rule them all! Commits landing asap.